src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py

```diff
@@ -42,13 +42,16 @@ class CMSSW(Executor):
 
     """
 
-    def _setStatus(self, returncode):
+    def _setStatus(self, returnCode, returnMessage):
         """
         Set return code.
         """
-        self.setCondorChirpAttrDelayed('Chirp_WMCore_cmsRun_ExitCode', returncode)
-        self.setCondorChirpAttrDelayed('Chirp_WMCore_%s_ExitCode' % self.stepName, returncode)
-        self.step.execution.exitStatus = returncode
+        self.setCondorChirpAttrDelayed('Chirp_WMCore_cmsRun_ExitCode', returnCode)
+        self.setCondorChirpAttrDelayed('Chirp_WMCore_%s_ExitCode' % self.stepName, returnCode)
+        if returnMessage:
```

Comment:
should we also check whether `returnCode != 0`? Such that we don't even chirp that attribute if the job didn't fail?

Reply:
I think the Report will always return a `None` return message if exit code is 0, but an extra safety check doesn't hurt.